querqy-core/src/main/java/querqy/rewrite/lookup/StateExchangingSequenceExtractor.java

```diff
@@ -0,0 +1,126 @@
+package querqy.rewrite.lookup;
+
+import lombok.RequiredArgsConstructor;
+import querqy.model.AbstractNodeVisitor;
+import querqy.model.BooleanClause;
+import querqy.model.BooleanQuery;
+import querqy.model.Term;
+import querqy.rewrite.lookup.model.Sequence;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
```

Comment:
What is "StateExchanging"? Something that applies a state change? As this class is not self explanatory, can you please add a short JavaDoc comment (and maybe rename it)?

Reply:
It exchanges a state with a StateExchangingCollector. The extractor needs to store a state resulting from a lookup in the collector for (probably multiple) subsequent lookups (for a certain sequence) again happening in the collector. Therefore, the extractor and the collector permanently exchange states. I can create a comment for this in the code.

Reply:
How about this: As far as I understand, to the 'outer world', the StateExchangingSequenceExtractor is the `SequenceExtractor` (the StateExchanging refers to an implementation detail that the 'outer world' might not be interested in). The `StateExchangingCollector` has more or less two functions: collect the matches and traverse the query graph to the next term. For me, this feels like a `TransducerAndCollector` (maybe without the `And`). 

Would it make sense to rename the classes/interfaces to  `SequenceExtractor` and `TransducerAndCollector`?

Reply:
The extractor traverses the graph and **administers** states.
The collector exposes seqs against the map (or maybe something else) and collects results by **using** the states.

Reply:
We finally agreed on 
* Dividing TrieMapStateExchangingCollector into two classes (TrieMapAutomatonWrapper & MatchCollector)
* Renaming StateExchangingCollector to AutomatonWrapper
* Renaming StateExchangingSequenceExtractor to AutomatonSequenceExtractor

This will happen in a subsequent PR.